tests/ai_reports/test_classification_signals.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+from django.db.models import signals
+from django.test import override_settings
+
+from adhocracy4.comments.models import Comment
+from apps.ai_reports.signals import get_ai_classification
+
+
+@pytest.mark.django_db
+def test_comment_not_sent_to_xai_no_url(mocker, idea, comment_factory, caplog):
+    task_mock = mocker.patch(
+        "apps.ai_reports.tasks.get_classification_for_comment" ".delay"
+    )
+    assert get_ai_classification in signals.post_save._live_receivers(Comment)
+    comment_factory(content_object=idea, comment="lala")
+    task_mock.assert_not_called()
+    assert len(caplog.records) == 1
+    assert "no xai api url provided" in str(caplog.records[-1])
+
+
+@override_settings(XAI_API_URL="https://liqd.net")
+@pytest.mark.django_db
+def test_comment_sent_to_xai_on_comment_text_change(mocker, idea, comment_factory):
+    task_mock = mocker.patch(
+        "apps.ai_reports.tasks.get_classification_for_comment" ".delay"
+    )
+
+    assert get_ai_classification in signals.post_save._live_receivers(Comment)
+    comment = comment_factory(content_object=idea, comment="lala")
+    task_mock.assert_called_once_with(comment.pk)
+    task_mock.reset_mock()
+
+    comment.comment = "modified comment"
+    comment.save()
+    task_mock.assert_called_once_with(comment.pk)
+
+
+@override_settings(XAI_API_URL="https://liqd.net")
+@pytest.mark.django_db
+def test_comment_not_sent_to_xai_without_comment_text_change(
+    mocker, idea, comment_factory, caplog
+):
+    task_mock = mocker.patch(
+        "apps.ai_reports.tasks.get_classification_for_comment" ".delay"
+    )
+    assert get_ai_classification in signals.post_save._live_receivers(Comment)
```

Comment:
not sure what you are testing here. Since it is a post save signal and it's calling other functions, then we need to save a comment, and see if the task_mock has been called.

Reply:
this line just asserts that `get_ai_classification` was correctly egistered as a post_save signal for the `Comment` class

Reply:
ah, if you were referring to the test in general, it tests that the comment isn't send to the xai again if something else than the text changed